www/src/pages/table/NodeSearchComponent.tsx

```diff
@@ -44,62 +55,121 @@ export const NodeSearchComponent = withStyles(styles)(
-          options.push({
-            type: "class",
-            id: r.id,
-            value: `class:${r.id}`,
-            label: sm.graph.uriCount.nextLabel(r.uri, r.readableLabel),
-          });
-        }
+        const options: SearchOptions[] = [];
 
         for (const u of sm.graph.nodes) {
+          if (
+            classSearchOnly &&
+            !(u.nodetype == "class_node" || u.nodetype == "literal_node")
+          )
+            continue;
           options.push({
             type: u.nodetype,
             id: u.id,
             value: `${u.nodetype}:${u.id}`,
             label: sm.graph.uriCount.label(u),
+            filterlabel: sm.graph.uriCount.label(u),
             className: classes[u.nodetype],
-          } as any);
+          });
         }
+
+        setSearchOptions(options);
+        console.log(options);
         return options;
-      }, [classStore.records.size]);
+      }, [sm.graph.version]);
 
       // search for additional values if it's not in the list
       const onSearch = (query: string) => {
-        if (query === "") return;
-        seqInvoker.exec(() => {
-          return classStore.fetchById(query).then(() => 1);
+        if (query === "") {
+          setSearchOptions(options);
+          return;
+        }
+        const loaderOption: SearchOptions = {
+          type: "class",
+          id: "",
+          label: <Spin style={{ width: "100%", marginTop: 3 }} size="large" />,
+          filterlabel: query,
+          value: "",
+          className: "",
+        };
+
+        console.log(options);
+        setSearchOptions([...options, loaderOption]);
+
+        classStore.findByName(query).then((data) => {
+          let searchResults: SearchOptions[] = data.map(
+            (searchResult: ClassTextSearchResult) => {
+              return {
+                type: "class",
+                id: searchResult.id,
+                label: (
+                  <SearchOptionsComponent
+                    id={searchResult.id}
+                    description={searchResult.description}
+                    label={searchResult.label}
+                  />
+                ),
+                filterlabel: `${searchResult.label} (${searchResult.id})`,
+                value: `class:${searchResult.id}`,
+              };
+            }
+          );
+          setSearchOptions([...options, ...searchResults]);
         });
       };
 
       return (
         <Select
           allowClear={true}
-          options={options}
-          optionFilterProp="label"
+          options={searchOptions}
+          onClear={() => setSearchOptions(options)}
+          optionFilterProp="filterlabel"
+          defaultActiveFirstOption={false}
           className={classes.selection}
           showSearch={true}
-          onSearch={onSearch}
+          filterOption={(inputValue, option) => {
+            if (option!.type != "class") {
+              let label = option?.filterlabel!;
```

Comment:
rewrite line 146 to 158 to something like: `return inputValue.split(" ").some(...)`;

Reply:
no question mark. also convert toLowerCase here.

Reply:
the replacement is still incorrect